### PR TITLE
Fix for verifier issue in 5.4 with clang-13

### PR DESCRIPTION
**Pull request by tpapagian**
FIXES: https://github.com/cilium/tetragon/issues/220

Tested over https://github.com/cilium/tetragon/pull/171 for clang-13 support.

Signed-off-by: Anastasios Papagiannis <tasos.papagiannnis@gmail.com>

### CODE REVIEW

**Comment by willfindlay** [comment]
Thanks for the fix! I'll have a go at testing this as well on Monday.

**Review by olsajiri — APPROVED** [review]

File: bpf/process/data_event.h [hunk 1]
```diff
@@ -6,30 +6,26 @@
...
-
 	/* Code movement from clang forces us to inline bounds checks here */
-	asm volatile("%[rd_bytes] &= 0x7fff;\n"
-		     "if %[rd_bytes] < 32736 goto +1\n;"
-		     "%[rd_bytes] = 32736;\n"
-		     :
-		     : [rd_bytes] "+r"(rd_bytes)
-		     :);
-	err = probe_read(&msg->arg[0], rd_bytes, (char *)uptr);
+	asm volatile goto("%[bytes] &= 0x7fff;\n"
```
Comment (kkourt) [inline bpf/process/data_event.h:12]: I'm reading the comment in: https://github.com/cilium/tetragon/blob/38d3a6135cf9e23265c4319c3507bfffb3b42a5f/bpf/lib/data_msg.h#L18-L28

which describes the `size &= 0x7fff` part. Is this still needed after this change?

Comment (tpapagian) [reply bpf/process/data_event.h:12]: Yes, you are correct. It seems that we can remove the `"%[bytes] &= 0x7fff;\n"` command as well. I will update the PR.

Comment (kkourt) [reply bpf/process/data_event.h]: Thanks!

How about we also add a:

```
#define MESSAGE_DATA_ARG_LEN 32736
```

And use that both in `struct msg_data` and the assembly code? (I think there should be a way to pass constants as arguments).

Comment (tpapagian) [reply bpf/process/data_event.h]: Done and tested in 4.19, 5.4, 5.10, and 5.13.

**Review by kkourt — APPROVED** [review]

**Comment by jrfastab** [comment]
ping @tpapagian would like to merge this if we can and get off clang fork. Can you take a look thanks!

**Pull request merged**